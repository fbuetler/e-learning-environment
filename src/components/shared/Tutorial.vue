<template>
  <div id="tutorial-wrapper">
    <div
      id="tutorial"
      class="flex-item flex-center card clickable responsive"
      title="Anleitung Ã¶ffnen"
      @click="showModal = true"
    >
      <img :src="require('@/assets/icons/lightbulb.png')" />
    </div>
    <modal v-if="showModal" @close="showModal = false">
      <h3 slot="header">Anleitung</h3>
      <div
        slot="body"
        class="flex-item flex-center flex-space-between flex-col"
      >
        <div class="flex-item flex-center flex-space-between flex-row">
          <div class="description equal-space">
            <slot name="description">Unintentionally empty!</slot>
          </div>
          <div class="responsive equal-space">
            <slot name="video">Unintentionally empty!</slot>
          </div>
        </div>
        <!-- 
        Uncomment the following code to show a button in the tutorial to start
        the tutorial animation. This is used to create the tutorial videos.
        <div>
          <button
            class="card clickable"
            @click="
              showModal = false;
              $emit('start-tutorial-animation');
            "
          >
            Live Demo!
          </button>
        </div>
        -->
      </div>
    </modal>
  </div>
</template>

<script lang="ts">
import { Vue, Component } from "vue-property-decorator";
import Modal from "@/components/shared/Modal.vue";

@Component<Tutorial>({
  components: {
    Modal,
  },
})
export default class Tutorial extends Vue {
  showModal = false;
}
</script>

<style scoped>
.description > p {
  margin: 1em;
  text-align: justify;
}
</style>
