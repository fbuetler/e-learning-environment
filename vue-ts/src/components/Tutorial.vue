<template>
  <div>
    <div
      id="tutorial"
      class="tutorial flex-item flex-center card"
      title="Anleitung Ã¶ffnen"
      @click="showModal = true"
    >
      <img :src="require('@/assets/icons/lightbulb.png')" />
    </div>
    <modal v-if="showModal" @close="showModal = false">
      <h3 slot="header">Anleitung</h3>
      <div
        slot="body"
        class="flex-item flex-center flex-space-between flex-row"
      >
        <div class="equal-space">
          <p v-html="description"></p>
        </div>
        <div class="equal-space">
          <video loop controls muted>
            <source :src="require('@/assets/tutorials/' + video)" />
            Your browser does not support the video tag.
          </video>
        </div>
      </div>
    </modal>
  </div>
</template>

<script lang="ts">
import { Vue, Component, Prop } from "vue-property-decorator";
import Modal from "./Modal.vue";

@Component<Tutorial>({
  components: {
    Modal,
  },
})
export default class Tutorial extends Vue {
  @Prop({ required: true })
  description: string;
  @Prop({ required: true })
  video: string;

  showModal = false;
}
</script>

<style scoped>
.tutorial > img {
  width: 100%;
  max-width: 30px;
}
p {
  margin: 1em;
  text-align: justify;
}
video {
  max-height: 100%;
  max-width: 100%;
}
</style>
