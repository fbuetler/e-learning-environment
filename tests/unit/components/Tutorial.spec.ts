import { mount } from "@vue/test-utils";
import Tutorial from "@/components/shared/Tutorial.vue";

describe("Tutorial.vue", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(Tutorial);
  });

  it("is a Vue instance", () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
