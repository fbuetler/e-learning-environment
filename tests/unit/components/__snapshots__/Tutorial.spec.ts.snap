// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tutorial.vue renders correctly 1`] = `
<div
  id="tutorial-wrapper"
>
  <div
    class="flex-item flex-center card clickable responsive"
    id="tutorial"
    title="Anleitung Ã¶ffnen"
  >
    <img
      src=""
    />
  </div>
   
  <!---->
</div>
`;
